tools/object_identifier.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"go/ast"
+	"go/token"
+	"html/template"
+	"log"
+	"sort"
+	"strings"
+)
+
+type identifiableObject struct {
+	ObjectName       string
+	IdentifyingField string
+}
+
+func (gen *ObjectGenerator) generateGetIdentifier() {
+	const templates = `
+{{range .IdentifiedObjects }}
+// GetIdentifier returns the primary identifier of a {{.ObjectName}} object
+func (x *{{.ObjectName}}) GetIdentifier(ctx context.Context) (string, error) {
```

Comment:
small nitpick: change the receivers name from `x` to `o`, which is the common single char name for objects